src/System.Device.Ports/System.Device.Ports.SerialPort/SerialPort.Settings.cs

```diff
@@ -0,0 +1,728 @@
+        /// </summary>
+        /// <param name="parity">The parity value to set</param>
+        protected internal abstract void SetParity(Parity parity);
+
+        /// <summary>
+        /// The data bits
+        /// </summary>
+        public int DataBits
+        {
+            get => _dataBits;
+            set
+            {
+                // 9 data bit is only supported by toggling the parity bit
+                if (_dataBits < MinDataBits || _dataBits > MaxDataBitsNoParity ||
+                    (_dataBits == MaxDataBitsNoParity && Parity != Parity.None))
+                {
+                    throw new ArgumentOutOfRangeException(nameof(DataBits), Strings.InvalidDataBits);
+                }
+
+                if (value == _dataBits)
+                {
+                    return;
+                }
+
+                _dataBits = value;
+                SetDataBits(_dataBits);
+            }
+        }
+
+        /// <summary>
+        /// Set the communication data bits
+        /// </summary>
+        /// <param name="dataBits">The data bits value to set</param>
+        protected internal abstract void SetDataBits(int dataBits);
+
+        /// <summary>
+        /// The stop bits
+        /// </summary>
+        public StopBits StopBits
+        {
+            get => _stopBits;
+            set
+            {
+                if (value < StopBits.One || value > StopBits.OnePointFive)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(StopBits), Strings.ArgumentOutOfRange_Enum);
+                }
+
+                if (value == _stopBits)
+                {
+                    return;
+                }
+
+                _stopBits = value;
+                SetStopBits(_stopBits);
+            }
+        }
+
+        /// <summary>
+        /// Set the communication stop bits
+        /// </summary>
+        /// <param name="stopBits">The stop bits value to set</param>
+        protected internal abstract void SetStopBits(StopBits stopBits);
+
+        /// <summary>
+        /// Gets or sets the break signal state.
+        /// </summary>
+        public bool BreakState
+        {
+            get => _breakState;
+            set
+            {
+                /*
+                if (value == _breakState)
+                {
+                    return;
+                }
+                */
+
+                _breakState = value;
+                SetBreakState(_breakState);
+            }
+        }
+
+        /// <summary>
+        /// Sets the break signal state.
+        /// </summary>
+        /// <param name="breakState">true if the port is in a break state; otherwise, false.</param>
+        protected internal abstract void SetBreakState(bool breakState);
+
+        /// <summary>
+        /// Gets the number of bytes of data in the receive buffer.
+        /// </summary>
+        public int BytesToRead
+        {
+            get
+            {
+                if (!_isOpen)
```

Comment:
should we have similar check everywhere?

Reply:
I have to see as soon as something starts working.
I am trying to port the same behavior of the `SerialPort` even if the code is different.